earthfile2llb/converter.go

```diff
@@ -1773,6 +1780,23 @@ func (c *Converter) internalRun(ctx context.Context, opts ConvertRunOpts) (pllb.
 	}
 }
 
+// secretID returns query parameter style string that contains the secret
+// version, name, org, and project. The version value informs the secret
+// providers how to handle the secret name and whether to use the new
+// project-based secrets endpoints.
+func (c *Converter) secretID(name string) string {
+	v := url.Values{}
+	v.Set("name", name)
+	if c.ftrs.UseProjectSecrets {
+		v.Set("v", "1")
+		v.Set("org", c.varCollection.Org())
+		v.Set("project", c.varCollection.Project())
+	} else {
+		v.Set("v", "0")
+	}
```

Comment:
The order of the query parameters is guaranteed to be consistent, I imagine? This is important for LLB/cache consistency.